src/client/localplayer.cpp

```diff
@@ -694,47 +667,55 @@ void LocalPlayer::applyControl(float dtime, Environment *env)
 
 	// Accelerate to target speed with maximum increment
 	accelerate((speedH + speedV) * physics_override_speed,
-			incH * physics_override_speed * slip_factor, incV * physics_override_speed,
-			pitch_move);
+		incH * physics_override_speed * slip_factor, incV * physics_override_speed,
+		pitch_move);
 }
 
 v3s16 LocalPlayer::getStandingNodePos()
 {
-	if(m_sneak_node_exists)
+	if (m_sneak_node_exists)
 		return m_sneak_node;
+
 	return m_standing_node;
 }
 
 v3s16 LocalPlayer::getFootstepNodePos()
 {
+	// Emit swimming sound if the player is in liquid
 	if (in_liquid_stable)
-		// Emit swimming sound if the player is in liquid
 		return floatToInt(getPosition(), BS);
+
+	// BS * 0.05 below the player's feet ensures a 1/16th height
+	// nodebox is detected instead of the node below it.
 	if (touching_ground)
-		// BS * 0.05 below the player's feet ensures a 1/16th height
-		// nodebox is detected instead of the node below it.
-		return floatToInt(getPosition() - v3f(0, BS * 0.05f, 0), BS);
+		return floatToInt(getPosition() - v3f(0.0f, BS * 0.05f, 0.0f), BS);
+
 	// A larger distance below is necessary for a footstep sound
 	// when landing after a jump or fall. BS * 0.5 ensures water
 	// sounds when swimming in 1 node deep water.
-	return floatToInt(getPosition() - v3f(0, BS * 0.5f, 0), BS);
+	return floatToInt(getPosition() - v3f(0.0f, BS * 0.5f, 0.0f), BS);
 }
 
 v3s16 LocalPlayer::getLightPosition() const
 {
-	return floatToInt(m_position + v3f(0,BS+BS/2,0), BS);
+	return floatToInt(m_position + v3f(0.0f, BS * 1.5f, 0.0f), BS);
 }
 
 v3f LocalPlayer::getEyeOffset() const
 {
-	float eye_height = camera_barely_in_ceiling ?
-		m_eye_height - 0.125f : m_eye_height;
-	return v3f(0, BS * eye_height, 0);
+	float eye_height = camera_barely_in_ceiling ? m_eye_height - 0.125f : m_eye_height;
+	return v3f(0.0f, BS * eye_height, 0.0f);
+}
+
+bool LocalPlayer::isDead() const
+{
+	FATAL_ERROR_IF(!getCAO(), "LocalPlayer's CAO isn't initialized");
```

Comment:
Why is the error needed? You could assume `false` as long it's not initialized.

Reply:
The only function that calls this is when the CAO might not be initialised yet is `Client::sendPlayerPos`, and that checks if the CAO has been initialised before calling it. This error is to prevent other functions in the future to call this before the CAO is initialised and end up hitting a segfault for no apparent reason.

OTOH, if this just returns false, it'd mean that all functions that call `isDead` would process code that is based on the player being alive, while the player _could_ be dead. This, again, has the chance to create similar problems, and segfault or not, it'd take quite a while to figure out why the player is said to be alive while they obviously aren't.